### PR TITLE
schema: support flashback with ddl hisotry

### LINKED ISSUE #6566: Tiflash support flashback cluster with ddl history
## Enhancement
ref https://github.com/pingcap/tidb/issues/40026, when tiflash applied a schema diff with `RegenerateSchemaMap` flag, it should reload all schema info from TiKV

### PR DESCRIPTION
**Pull request by Defined2014**
### What problem does this PR solve?

Issue Number: close #6566

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
```
use tiup to create a cluster with tiflash, then create a table with tiflash replica. TiFlash works well after flashback cluster operate.
```
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by Defined2014** [comment]
@zanmato1984 @windtalker @gengliqi PTAL

File: dbms/src/TiDB/Schema/TiDBSchemaSyncer.h [hunk 1]
```diff
@@ -213,6 +213,14 @@ struct TiDBSchemaSyncer : public SchemaSyncer
                     LOG_WARNING(log, "Skip the schema diff from version {}. ", cur_version + diff_index + 1);
                     continue;
                 }
+
+                if (schema_diff->regenerate_schema_map)
+                {
+                    // If `schema_diff.regenerate_schema_map` == true, return `-1` direclty, let TiFlash reload schema info from TiKV.
```
Comment (hongyunyan) [inline dbms/src/TiDB/Schema/TiDBSchemaSyncer.h:219]: Please also add related comments in L163 and L126, to make the code more easily to understand.

**Review by hongyunyan — COMMENTED** [review]
Could you please add an integration tests under tests/fullstack-test2/ddl

**Review by CalvinNeo — APPROVED** [review]
lgtm

File: dbms/src/TiDB/Schema/TiDBSchemaSyncer.h (hunk 1 above)
Comment (Defined2014) [reply dbms/src/TiDB/Schema/TiDBSchemaSyncer.h:220]: L163 updated, L126 is not related with `flashback`.

**Review by hongyunyan — APPROVED** [review]
LGTM

**Pull request merged**